tests.py

```diff
@@ -436,40 +546,239 @@ def __getitem__(self, key):
+
+    def test_get_qr_code_image(self):
+        response = {'id': 'id',
+                    'enabled': True,
+                    'pending': True,
+                    'user': {'id': 'id', 'name': 'name'}}
+        pairing = toopher.Pairing(response)
+
+        api = toopher.ToopherApi('key', 'secret')
+        with open('qr_image.png', 'rb') as qr_image:
+            api.client = HttpClientMock({
+                'qr/pairings/{}'.format(pairing.id): (200, qr_image.read())})
+
+            qr_image_data = pairing.get_qr_code_image(api)
+            self.assertEqual(api.client.last_called_method, 'GET')
+            with open('new_image.png', 'wb') as new_image:
+                new_image.write(qr_image_data)
+
+            im = Image.open('new_image.png')
+            try:
+                im.verify()
+            except:
+                self.assertTrue(False)
+            os.remove('new_image.png')
+
+
+class UserTerminalTests(unittest.TestCase):
+    def test_incomplete_response_raises_exception(self):
+        response = {'key': 'value'}
+        def fn():
+            toopher.UserTerminal(response)
+        self.assertRaises(toopher.ToopherApiError, fn)
+
+    def test_refresh_from_server(self):
+        response = {'id': 'id',
+                    'name': 'name',
+                    'name_extra': 'name_extra',
+                    'user': {'id': 'id', 'name': 'name'}}
+        user_terminal = toopher.UserTerminal(response)
+        api = toopher.ToopherApi('key', 'secret')
+        api.client = HttpClientMock({
+            'user_terminals/{0}'.format(user_terminal.id): (200,
+                json.dumps({'id': 'id',
+                    'name': 'name changed',
+                    'name_extra': 'name_extra changed',
+                    'user': {'id': 'id', 'name': 'name changed'}}))})
+        user_terminal.refresh_from_server(api)
+        self.assertEqual(api.client.last_called_method, 'GET')
+
+
+        self.assertEqual(user_terminal.id, "id")
+        self.assertEqual(user_terminal.name, "name changed")
+        self.assertEqual(user_terminal.name_extra, "name_extra changed")
+        self.assertEqual(user_terminal.user_name, "name changed")
+        self.assertEqual(user_terminal.user_id, "id")
+
+
+class UserTests(unittest.TestCase):
+    def test_incomplete_response_raises_exception(self):
+        response = {'key': 'value'}
+        def fn():
+            toopher.User(response)
+        self.assertRaises(toopher.ToopherApiError, fn)
+
+    def test_refresh_from_server(self):
+        response = {'id': 'id',
+                    'name': 'name',
+                    'disable_toopher_auth': False}
+        user = toopher.User(response)
+        api = toopher.ToopherApi('key', 'secret')
+        api.client = HttpClientMock({
+            'users/{0}'.format(user.id): (200,
+                json.dumps({'id': 'id',
+                            'name':'name CHANGED',
+                            'disable_toopher_auth':'true'}))})
```

Comment:
It's not a release blocker, but I'd like us to consistently use this style of creating the response payload in tests.

```
'{"id":"1", "name":"name", "name_extra":"name_extra", "user":{"id":"1","name":"some user"}}'
```
Would become something like this:
```
json.dumps({
  "id": "1",
  "name": "name",
  "name_extra": "name_extra",
  "user": {
    "id": "1",
    "name": "some user"
  }
})'
```